diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -62,10 +62,15 @@ if(MSVC)
   set(Boost_USE_STATIC_RUNTIME ON)
 endif()
 
-if(LINUX)
+if(GANNYU_MOBILE_USE_STD_REGEX)
+  set(Boost_FOUND TRUE)
+  set(Boost_INCLUDE_DIRS ${Boost_INCLUDE_DIR})
+  set(Boost_LIBRARY_DIRS "")
+  set(Boost_LIBRARIES "")
+elseif(LINUX)
   find_package(Boost 1.74.0 REQUIRED COMPONENTS regex)
 else()
-  find_package(Boost 1.77.0)
+  find_package(Boost 1.77.0 REQUIRED)
 endif()
 if(Boost_FOUND)
   include_directories(${Boost_INCLUDE_DIRS})
diff --git a/src/rime/algo/algebra.cc b/src/rime/algo/algebra.cc
--- a/src/rime/algo/algebra.cc
+++ b/src/rime/algo/algebra.cc
@@ -68,7 +68,7 @@ bool Projection::Load(an<ConfigList> settings) {
     an<Calculation> x;
     try {
       x.reset(calc.Parse(formula));
-    } catch (boost::regex_error& e) {
+    } catch (std::regex_error& e) {
       LOG(ERROR) << "Error parsing formula '" << formula << "': " << e.what();
     }
     if (!x) {
diff --git a/src/rime/algo/calculus.cc b/src/rime/algo/calculus.cc
--- a/src/rime/algo/calculus.cc
+++ b/src/rime/algo/calculus.cc
@@ -112,7 +112,7 @@ Calculation* Transformation::Parse(const vector<string>& args) {
 bool Transformation::Apply(Spelling* spelling) {
   if (!spelling || spelling->str.empty())
     return false;
-  string result = boost::regex_replace(spelling->str, pattern_, replacement_);
+  string result = std::regex_replace(spelling->str, pattern_, replacement_);
   if (result == spelling->str)
     return false;
   spelling->str.swap(result);
@@ -135,7 +135,7 @@ Calculation* Erasion::Parse(const vector<string>& args) {
 bool Erasion::Apply(Spelling* spelling) {
   if (!spelling || spelling->str.empty())
     return false;
-  if (!boost::regex_match(spelling->str, pattern_))
+  if (!std::regex_match(spelling->str, pattern_))
     return false;
   spelling->str.clear();
   return true;
diff --git a/src/rime/algo/calculus.h b/src/rime/algo/calculus.h
--- a/src/rime/algo/calculus.h
+++ b/src/rime/algo/calculus.h
@@ -9,7 +9,7 @@
 #define RIME_CALCULUS_H_
 
 #include <stdint.h>
-#include <boost/regex.hpp>
+#include <regex>
 #include <rime_api.h>
 #include <rime/common.h>
 #include "spelling.h"
@@ -54,7 +54,7 @@ class Transformation : public Calculation {
   bool Apply(Spelling* spelling) override;
 
  protected:
-  boost::regex pattern_;
+  std::regex pattern_;
   string replacement_;
 };
 
@@ -66,7 +66,7 @@ class Erasion : public Calculation {
   bool addition() override { return false; }
 
  protected:
-  boost::regex pattern_;
+  std::regex pattern_;
 };
 
 // derive/x/X/
diff --git a/src/rime/algo/encoder.cc b/src/rime/algo/encoder.cc
--- a/src/rime/algo/encoder.cc
+++ b/src/rime/algo/encoder.cc
@@ -96,7 +96,7 @@ bool TableEncoder::LoadSettings(Config* config) {
       auto pattern = As<ConfigValue>(*it);
       if (!pattern)
         continue;
-      exclude_patterns_.push_back(boost::regex(pattern->str()));
+      exclude_patterns_.push_back(std::regex(pattern->str()));
     }
   }
   config->GetString("encoder/tail_anchor", &tail_anchor_);
@@ -131,8 +131,8 @@ bool TableEncoder::ParseFormula(const string& formula,
 }
 
 bool TableEncoder::IsCodeExcluded(const string& code) {
-  for (const boost::regex& pattern : exclude_patterns_) {
-    if (boost::regex_match(code, pattern))
+  for (const std::regex& pattern : exclude_patterns_) {
+    if (std::regex_match(code, pattern))
       return true;
   }
   return false;
diff --git a/src/rime/algo/encoder.h b/src/rime/algo/encoder.h
--- a/src/rime/algo/encoder.h
+++ b/src/rime/algo/encoder.h
@@ -8,7 +8,7 @@
 #ifndef RIME_ENCODER_H_
 #define RIME_ENCODER_H_
 
-#include <boost/regex.hpp>
+#include <regex>
 #include <rime_api.h>
 
 namespace rime {
@@ -92,7 +92,7 @@ class RIME_DLL TableEncoder : public Encoder {
   bool loaded_;
   // settings
   vector<TableEncodingRule> encoding_rules_;
-  vector<boost::regex> exclude_patterns_;
+  vector<std::regex> exclude_patterns_;
   string tail_anchor_;
   // for optimization
   int max_phrase_length_;
