diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b67549e..afbd643 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -208 +208,3 @@
-add_subdirectory(tools)
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
+  add_subdirectory(tools)
+endif()
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -223,7 +223,9 @@
 add_subdirectory(src)
 add_subdirectory(doc)
-add_subdirectory(data)
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
+  add_subdirectory(data)
+endif()
 add_subdirectory(test)
